### PR TITLE
Feature/rht 1874 automatic update position in sf

### PR DESCRIPTION
**Pull request by ntqdinh-axonivy**

### CODE REVIEW

File: successfactors-connector/config/rest-clients.yaml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
   #Service for namespace SFODataSet
   #This service is located at [https://localhost/service-root/](https://localhost/service-root/)
-  SuccessFactors:
+  SuccessFactors (Service for namespace SFODataSet):
     UUID: 4c029770-6b05-4cf7-9b23-b2434bb500d6
     Url: https://{host}/odata/v2
     Features:
     - com.axonivy.connector.successfactors.connector.rest.odata.OdataJsonFeature
     - ch.ivyteam.ivy.rest.client.authentication.HttpBasicAuthenticationFeature
+    - ch.ivyteam.ivy.rest.client.mapper.JsonFeature
```
Comment (ndkhanh-axonivy) [inline successfactors-connector/config/rest-clients.yaml:10]: remove

File: successfactors-connector/src/com/axonivy/connector/successfactors/connector/rest/odata/OdataJsonFeature.java [hunk 2]
```diff
@@ -96,16 +94,17 @@ public ObjectMapper locateMapper(Class<?> type, MediaType mediaType) {
 		 */
 		protected JsonNode manipulateJson(JsonNode node) {
 			if (node.has(Field.VALUE)) {
-				node = node.get(Field.VALUE);
+				if (node.get(Field.VALUE).has(Field.RESULT)) {
+					node = node.get(Field.VALUE);
+				}
 			}
-			if(node.has("results")) {
+			if (node.has("results") || node.has(Field.VALUE)) {
```
Comment (phhung-axonivy) [inline successfactors-connector/src/com/axonivy/connector/successfactors/connector/rest/odata/OdataJsonFeature.java:101]: node.has("results") -> node.has(Field.RESULT)

File: successfactors-connector/src/com/axonivy/connector/successfactors/connector/rest/odata/SuccessFactorsTypeCustomizations.java [hunk 3]
```diff
@@ -2,286 +2,351 @@
...
-	}
-
-	@SuppressWarnings({"unchecked", "rawtypes"})
-	private void nullify(Class<?> raw) {
-		addDeserializer(raw, new Nullifier(raw));
+	public SuccessFactorsTypeCustomizations() {
+		addSerializer(SFODataCustomNavigationCreate.class, new CustomNavigationCreateSerializer());
+		addSerializer(SFODataPositionUpsert.class, new PositionUpsertSerializer());
+		addSerializer(SFODataFOJobCodeUpsert.class, new FOJobCodeUpsertSerializer());
+//		addDeserializer(AnyOfSFODataEmpJobSeqNumber.class, new SquNumberDeserializer());
```
Comment (phhung-axonivy) [inline successfactors-connector/src/com/axonivy/connector/successfactors/connector/rest/odata/SuccessFactorsTypeCustomizations.java:32]: remove comment lines

File: successfactors-connector/src/com/axonivy/connector/successfactors/connector/rest/odata/SuccessFactorsTypeCustomizations.java [hunk 4]
```diff
@@ -2,286 +2,351 @@
...
-		}
-
-		@Override
-		public String toString() {
-			return value;
-		}
-
-	}
-	
+//	@SuppressWarnings({ "unchecked", "rawtypes" })
```
Comment (phhung-axonivy) [inline successfactors-connector/src/com/axonivy/connector/successfactors/connector/rest/odata/SuccessFactorsTypeCustomizations.java:122]: remove comment lines

**Review by phhung-axonivy — APPROVED** [review]
Looks good

**Pull request merged**